Sources/XcodeProjMapper/Utilities/DeveloperDirectoryProvider.swift

```diff
@@ -0,0 +1,26 @@
+import Command
+import Foundation
+import Path
+
+/// A protocol that obtains the current developer directory (via `xcode-select -p`) asynchronously.
+public protocol DeveloperDirectoryProviding {
+    /// Returns the absolute path to the currently selected Xcode’s Developer directory.
+    /// - Throws: If `xcode-select -p` fails or if the output is invalid.
+    func developerDirectory() async throws -> AbsolutePath
+}
+
+public struct DeveloperDirectoryProvider: DeveloperDirectoryProviding {
```

Comment:
I wouldn't make this public as it's not expected that `XcodeProjMapper` would suddenly be a utility module for services like this one.